hyperspy/tests/utils/test_roi.py

```diff
@@ -112,95 +112,108 @@ def test_circle_spec(self):
         s = self.s_s
         s.data = np.ones_like(s.data)
         r = CircleROI(20, 25, 20)
+        r_ann = CircleROI(20, 25, 20, 15)
         sr = r(s)
+        sr_ann = r_ann(s)
         scale = s.axes_manager[0].scale
         n = int(round(40 / scale))
         nt.assert_equal(sr.axes_manager.navigation_shape, (n, n))
+        nt.assert_equal(sr_ann.axes_manager.navigation_shape, (n, n))
         # Check that mask is same for all images:
         for i in xrange(n):
             for j in xrange(n):
                 nt.assert_true(np.all(sr.data.mask[j, i, :] == True) or
                                np.all(sr.data.mask[j, i, :] == False))
+                nt.assert_true(np.all(sr_ann.data.mask[j, i, :] == True) or
+                               np.all(sr_ann.data.mask[j, i, :] == False))
         # Check that the correct elements has been masked out:
         mask = sr.data.mask[:, :, 0]
         print mask   # To help debugging, this shows the shape of the mask
         np.testing.assert_array_equal(
             np.where(mask.flatten())[0],
-            [0,  1,  6,  7,  8, 15, 48, 55, 56, 57, 62, 63])
+            [0, 1, 6, 7, 8, 15, 48, 55, 56, 57, 62, 63])
+        mask_ann = sr_ann.data.mask[:, :, 0]
+        print mask_ann   # To help debugging, this shows the shape of the mask
+        np.testing.assert_array_equal(
+            np.where(mask_ann.flatten())[0],
+            [0, 1, 6, 7, 8, 10, 11, 12, 13, 15, 17, 18, 19, 20, 21, 22, 25,
+             26, 27, 28, 29, 30, 33, 34, 35, 36, 37, 38, 41, 42, 43, 44, 45, 46,
+             48, 50, 51, 52, 53, 55, 56, 57, 62, 63])
         # Check that mask works for sum
-        nt.assert_equal(np.sum(sr.data), (n**2 - 3*4)*4)
+        nt.assert_equal(np.sum(sr.data), (n**2 - 3 * 4) * 4)
+        nt.assert_equal(np.sum(sr_ann.data), 4 * 5 * 4)
```

Comment:
Haven't run the code, but shouldn't this sum be `4 * 5 * 4 * 4`? Navigation dims are 4x4, and each mask has four quadrants with five unmasked entries each.